### PR TITLE
Improve Test Coverage for packages/web3/src/token

### LINKED ISSUE #455: Improve Test Coverage for `packages/web3/src/token`
This issue aims to improve the test coverage for this folder by:  
1. Identifying untested or under-tested functions and classes.  
2. Writing additional unit tests to cover edge cases and scenarios.

### PR DESCRIPTION
**Pull request by No-bodyq**
resolves #455

### CODE REVIEW

File: packages/web3/src/token/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+// eslint-disable-next-line header/header
+import * as IndexExports from './index'
+import * as NFTModule from './nft'
+
+describe('index.ts exports', () => {
```
Comment (h0ngcha0) [inline packages/web3/src/token/index.test.ts:8]: I don't think this test is needed. The only thing in `index.ts` is  `export * from './nft'`

**Comment by No-bodyq** [comment]
Okay 
I'll take it out

File: packages/web3/src/token/nft.test.ts [hunk 2]
```diff
@@ -18,116 +18,127 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
...
-      description: '',
-      url: 'https://example.com/'
-    }
-    expect(() => validateNFTTokenUriMetaData(withWrongField)).toThrow(Error)
-    expect(() => validateNFTCollectionUriMetaData(withWrongField)).toThrow(Error)
+    it('validates metadata without attributes', () => {
+      expect(validateNFTTokenUriMetaData(validMetadataWithoutAttributes)).toEqual(validMetadataWithoutAttributes)
+    })
+
+    it('validates metadata with attributes', () => {
```
Comment (h0ngcha0) [inline packages/web3/src/token/nft.test.ts:44]: I am not sure if it adds that much clarity with `it('description....` around every tests, it only makes the diff unnecessarily large.

e.g. it is clear that this tests the case where we have empty description.
```
expect(() => validateNFTTokenUriMetaData(withEmptyDescription)).toThrow(Error)
```

Is it possible to not change the structure of the existing tests and only add/update more tests?
